src/Components/Search/SearchBar.tsx

```diff
@@ -127,13 +135,8 @@ export class SearchBar extends Component<Props, State> {
       null,
       error => {
         if (error) logger.error(error)
-        if (tracking) {
-          const edges = get(viewer, v => v.search.edges, [])
-          const action = `Auto-suggest with ${
-            edges.length === 0 ? "no results" : "results"
-          }`
-          tracking.trackEvent({ action, term })
-        }
+        const edges = get(viewer, v => v.search.edges, [])
+        this.trackSearch(term, edges.length > 0)
```

Comment:
Btw, this is currently not taking into account if an error occurred, is that correct? If you do need to, a pattern we’ve used in the past is to create a `handleSearchResult(error)` method, which in its body does the normal business logic and in its decorator handles the analytics.

Reply:
oh true, in the event of an error we _wouldnt_ want to fire these analytics.

In that case, yes I'll just do that.